### PR TITLE
Update create-react-app to 2.1.2

### PR DESCRIPTION
**Pull request by tiffon**
Update [create-react-app](https://github.com/facebook/create-react-app/blob/master/CHANGELOG.md) to 2.1.2 from 1.x.

Involved updating react-app-rewired: timarney/react-app-rewired#346.

More than a few files are changed due to prettier formatting. The first commit is the changes necessary to update to 2.1.2.

### CODE REVIEW

**Comment by yurishkuro** [comment]
could we not separate prettier formatting to a different PR?

**Comment by yurishkuro** [comment]
what is CRA 2 ?

**Comment by tiffon** [comment]
CRA is create-react-app, and CRA 2 is version >= 2.

We can separate the prettier formatting into a separate PR, I was thinking having separate commits is sufficient distinction.

**Comment by yurishkuro** [comment]
> I was thinking having separate commits is sufficient distinction.

once merged it will be squashed into one commit, which will be difficult to review if needed later.

File: packages/jaeger-ui/package.json [hunk 1]
```diff
@@ -4,28 +4,19 @@
...
     "enzyme-to-json": "^3.3.0",
+    "http-proxy-middleware": "^0.19.1",
     "less-vars-to-js": "^1.2.1",
     "react-app-rewire-babel-loader": "^0.1.1",
-    "react-app-rewire-less": "^2.1.0",
-    "react-app-rewired": "^1.4.0",
-    "react-scripts": "^1.0.11",
+    "react-app-rewire-less": "2.1.3",
+    "react-app-rewired": "2.0.2-next.0",
+    "react-scripts": "2.1.2",
```
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/jaeger-ui/package.json]: Lately, I've been going for exact, but it's not a big deal as we're using a `yarn.lock` file. Seems more important when publishing npm packages.

File: packages/jaeger-ui/package.json [hunk 2]
```diff
@@ -96,5 +86,6 @@
       "!src/utils/test/**/*.js",
       "!src/demo/**/*.js"
     ]
-  }
+  },
+  "browserslist": [">0.2%", "not dead", "not ie <= 11", "not op_mini all"]
```
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/jaeger-ui/package.json]: I increased to 0.5%:

```diff
 and_chr 70
 and_uc 11.8
-android 4.4.3-4.4.4
-android 4.2-4.3
 chrome 70
 chrome 69
-chrome 68
-chrome 67
-chrome 63
 chrome 49
 edge 17
 firefox 63
-firefox 52
 ios_saf 12.0-12.1
 ios_saf 11.3-11.4
 ios_saf 11.0-11.2
-ios_saf 10.3
-ios_saf 10.0-10.2
-ios_saf 9.3
 opera 56
 safari 12
-safari 11.1
-safari 10.1
 samsung 4
```

File: packages/jaeger-ui/src/model/search.js [hunk 3]
```diff
@@ -32,6 +32,7 @@ const comparators = {
  * @param  {Trace[]} traces The `Trace` array to sort.
  * @param  {string} sortBy A sort specification, see ./order-by.js.
  */
+// eslint-disable-next-line import/prefer-default-export
```
[the remark this answers is not in the document]
Comment (tiffon) [reply packages/jaeger-ui/src/model/search.js]: I think it only comes up when there is only one export in a file.

**Pull request merged**